client/my-sites/importer/newsletter/importer.tsx

```diff
@@ -123,7 +123,7 @@ export default function NewsletterImporter( {
 	useEffect( () => {
 		if ( urlData?.platform === engine ) {
 			if ( selectedSite && shouldResetImport && validFromSite === false ) {
-				resetPaidNewsletter( selectedSite.ID, engine, stepSlugs[ 0 ] );
+				resetPaidNewsletter( selectedSite.ID, engine, stepSlugs[ 0 ], fromSite );
```

Comment:
The URL is already being:
- Validated and analyzed by useAnalyzeUrlQuery
- Parsed and normalized in select-newsletter-form.tsx